### ISSUE TITLE
Add `--optimized` mode

### ISSUE THREAD

**Issue by igorbrigadir** [DocNow/twarc-csv#55]
The current output favors preserving as much information as possible from the original json, but there is some duplication, and a bunch of columns can be removed as they're rarely super useful.

The new `--optimized` mode will generate CSVs that drop a bunch of columns to save space:

```
edit_controls.edits_remaining
edit_controls.editable_until
entities.cashtags
entities.hashtags
entities.mentions
withheld.scope
withheld.copyright
author.id
author.entities.description.cashtags
author.entities.description.hashtags
author.entities.description.mentions
author.url
author.withheld.scope
author.withheld.copyright
geo.coordinates.coordinates
geo.coordinates.type
geo.country
geo.full_name
geo.geo.type
matching_rules
__twarc.retrieved_at
__twarc.url
__twarc.version
```

(exact list to be revised later)

These are the most commonly not present or duplicate ones, where the missing data can be inferred from the columns left over, or with the cashtags, hashtags, mentions, with twitter-text for example.

**Comment by edsu** [comment]
I'm interested in hearing where the need for this optimization arose. Was it a problem generating the CSV, or reading the generated CSV in another application? It sounds like the latter?

**Comment by igorbrigadir** [comment]
Just trying to deduplicate columns and remove mostly empty ones, so more can fit into memory, and other tools like [great expectations](https://docs.greatexpectations.io/docs/) or [pandas profiling](https://pandas-profiling.ydata.ai/docs/master/index.html) have an easier time.

**Comment by edsu** [comment]
Would being able to write to parquet help in situations like that?

**Comment by igorbrigadir** [comment]
Yep! Definitely i think #29 goes hand in hand with this - I think all of these things are basically the same task for me to do lol
